aggregate/aggregator.go

```diff
@@ -19,10 +19,25 @@ var (
 const (
 	maximumRecordSize       = 1024 * 1024 // 1 MB
 	defaultMaxAggRecordSize = 20 * 1024   // 20K
-	pKeyIdxSize             = 8
-	aggProtobufBytes        = 2 // Marshalling the data into protobuf adds an additional 2 bytes.
+	initialAggRecordSize    = 0
+	fieldNumberSize         = 1 // All field numbers are below 16, meaning they will only take up 1 byte
 )
 
+// Effectively just ceil(log base 128 of int)
+// The size in bytes that the protobuf representation will take
+func varint64Size(varint uint64) (size int) {
+	size = 1
+	for varint >= 0x80 {
```

Comment:
For learning purpose, is there any difference if we avoid the hex format, like `for varint >= 128`?

Reply:
nope, there's no difference :)